updater/lib/dependabot/analyse_files_command.rb

```diff
@@ -0,0 +1,152 @@
+        end
+
+        # TODO(brrygrdn): Remove?
+        #
+        # We submit this for historic instrumentation reasons in the update job, but it doubles up on
+        # this command's main purpose, so I think we should remove it - but I'd appreciate a Dependabot
+        # reviewer's eye on whether keeping this has benefits I'm not thinking of.
+        #
+        # service.update_dependency_list(dependency_snapshot: dependency_snapshot)
+
+        submission = GithubApi::DependencySubmission.new(
+          job: job,
+          snapshot: dependency_snapshot
+        )
+
+        Dependabot.logger.info("Dependency submission payload:\n#{JSON.pretty_generate(submission.payload)}")
+
+        # For now, we require the experiment to actually submit data as this alters repository dependency state
+        # so we should be very intentional in the event this is called by accident.
+        if Dependabot::Experiments.enabled?(:enable_dependency_submission_poc)
+          service.create_dependency_submission(dependency_submission: submission)
+        end
+
+        service.mark_job_as_processed(dependency_snapshot.base_commit_sha)
+      end
+    end
+
+    private
+
+    sig { returns(Dependabot::Job) }
+    def job
+      @job ||= T.let(Job.new_update_job(
+                       job_id: job_id,
+                       job_definition: Environment.job_definition,
+                       repo_contents_path: Environment.repo_contents_path
+                     ), T.nilable(Dependabot::Job))
+    end
+
+    sig { returns(String) }
+    def base_commit_sha
+      Environment.job_definition["base_commit_sha"]
+    end
+
+    sig { params(error: StandardError).void }
+    def handle_parser_error(error) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/PerceivedComplexity
+      # This happens if the repo gets removed after a job gets kicked off.
+      # The service will handle the removal without any prompt from the updater,
+      # so no need to add an error to the errors array
+      return if error.is_a? Dependabot::RepoNotFound
+
+      error_details = Dependabot.parser_error_details(error)
+
+      error_details ||=
+        # Check if the error is a known "run halting" state we should handle
+        if (error_type = Updater::ErrorHandler::RUN_HALTING_ERRORS[error.class])
+          { "error-type": error_type }
+        elsif error.is_a?(ToolVersionNotSupported)
+          Dependabot.logger.error(error.message)
+          {
+            "error-type": "tool_version_not_supported",
+            "error-detail": {
+              "tool-name": error.tool_name,
+              "detected-version": error.detected_version,
+              "supported-versions": error.supported_versions
+            }
+          }
+        else
+          # If it isn't, then log all the details and let the application error
+          # tracker know about it
+          Dependabot.logger.error error.message
```

Comment:
Will these get tagged with an `analyse_files` namespace or whatever from inside our instrumentation?

I suddenly see the importance of the `ERROR_TYPE_LABEL` 🤔. It'd be great to be able to filter these errors out from the get go.

Reply:
At the moment, they'll just associate with the job id - I suspect there is a job type pulled from the job definitions _somewhere_ that indicates whether it's a version- or security-update right now.

We could add a `command` tag early in the process so logs are partitioned by `update_files` or `update_graph` so we can filter.